### PR TITLE
Update JsonAccessor to support additions (binary data, Nan, Infinity)

### PR DESCRIPTION
**Pull request by ryanmelt**

### CODE REVIEW

File: openc3/lib/openc3/accessors/json_accessor.rb [hunk 1]
```diff
@@ -43,7 +51,7 @@ def self.write_item(item, value, buffer)
 
       # Update buffer
       if String === buffer
-        buffer.replace(JSON.generate(decoded, :allow_nan => true))
+        buffer.replace(JSON.generate(decoded.as_json, :allow_nan => true))
```
Comment (jmthomas) [inline openc3/lib/openc3/accessors/json_accessor.rb:54]: Create additions?

**Review by jmthomas — COMMENTED** [review]
Is this to fix a specific issue?

File: openc3/lib/openc3/accessors/json_accessor.rb (hunk 1 above)
Comment (ryanmelt) [reply openc3/lib/openc3/accessors/json_accessor.rb:54]: Not necessary on generate, only on parse

File: openc3/lib/openc3/accessors/json_accessor.rb [hunk 2]
```diff
@@ -18,8 +18,16 @@
 
 require 'json'
 require 'jsonpath'
+require 'openc3/io/json_rpc'
 require 'openc3/accessors/accessor'
 
+# Monkey patch JsonPath to enable create_additions and allow_nan to support binary strings, and NaN, Infinity, -Infinity
+class JsonPath
+  def self.process_object(obj_or_str, opts = {})
```
[the remark this answers is not in the document]
Comment (ryanmelt) [reply openc3/lib/openc3/accessors/json_accessor.rb:26]: its used internally in JsonPath.on

**Comment by ryanmelt** [comment]
Fixes specific issue that one of our customers was running into dealing with binary data in JSON

**Pull request merged**